jwst/cube_build/src/cube_match_internal.c

```diff
@@ -0,0 +1,433 @@
+  axis 2 of IFU cube
+crval_along : double
+  along slice reference value in IFU cube
+cdelt_along : double
+  along slice sampling in IFU cube
+crval3 : double
+   wavelength reference value in IFU cube
+cdetl3 : double
+   wavelength sampling in IFU cube
+a1, a2, a3, a4: double array 
+    array of corners of pixels holding along slice coordinates
+    a1 holds corner 1 
+    a2 holds corner 2 
+    a3 holds corner 3 
+    a4 holds corner 4 
+lam1, lam2, lam3, lam4: double array
+    array of corners of pixels holding wavelength coordinates
+    lam1 holds corner 1 
+    lam2 holds corner 2 
+    lam3 holds corner 3 
+    lam4 holds corner 4 
+acoord : double array
+   Array holds along slice coordinates in IFU cube
+zcoord : double array
+   Array holds the wavelength coordinates in the IFU cube
+ss : double array
+   Array holds the slice number
+pixel_flux : double array
+  Array that holds detector pixel flux
+pixel_error : double array
+  Array that holds detector pixel flux error
+
+Returns
+-------
+
+spaxel_flux : array
+spaxel_weight : array
+spaxel_iflux : array
+spaxel_var : array
+spaxel_dq : array
+
+*/
+
+#include <stdlib.h>
+#include <math.h>
+#include <stdio.h>
+#include <Python.h>
+#include <stdbool.h>
+#include <numpy/arrayobject.h>
+#include <numpy/npy_math.h>
+
+#define PY_ARRAY_UNIQUE_SYMBOL _jwst_cube_match_sky_numpy_api    //WHAT IS THIS AND WHERE IS IT USED???
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+
+
+// routines used from cube_utils.c
+extern double sh_find_overlap(const double xcenter, const double ycenter, 
+		      const double xlength, const double ylength,
+			      double xPixelCorner[],double yPixelCorner[]);
+
+
+extern double find_area_quad(double MinX, double MinY, double Xcorner[], double Ycorner[]);
+
+
+extern int alloc_flux_arrays(int nelem, double **fluxv, double **weightv, double **varv,  double **ifluxv);
+
+
+//_______________________________________________________________________
+//  based on a 2-D drizzling type of algorithm find the contribution of
+// each detector flux to the IFU cube
+//_______________________________________________________________________
+
+int match_detector_cube(int instrument, int naxis1, int naxis2, int nz, int npt, int ncube, int na,
+			double crval_along, double cdelt_along, double crval3, double cdelt3,
+			double *a1, double *a2,double *a3, double*a4,
+			double *lam1, double *lam2, double*lam3, double *lam4,
+			double *acoord, double *zcoord, int ss,
+			double *pixel_flux, double *pixel_err,
+			double **spaxel_flux, double **spaxel_weight, double **spaxel_var,double **spaxel_iflux){
+
+  double *fluxv, *weightv, *varv, *ifluxv ;  // vectors for spaxel 
+
+  // allocate memory to hold output 
+  if (alloc_flux_arrays(ncube, &fluxv, &weightv, &varv, &ifluxv)) return 1;
+    
+
+  // loop over each valid point on detector and find match to IFU cube based
+  // on along slice coordinate and wavelength
+  for (int ipixel= 0; ipixel< npt; ipixel++){
```

Comment:
@jemorrison I believe you need to declare the variables in the beginning of the file as certain compilers won't like the definition within the for loop.